tslearn/nonmyopic.py

```diff
@@ -0,0 +1,509 @@
+from sklearn.neural_network import MLPClassifier
+from sklearn.metrics import confusion_matrix
+from sklearn.base import BaseEstimator, ClassifierMixin
+from tslearn.utils import to_time_series_dataset
+from tslearn.clustering import TimeSeriesKMeans, silhouette_score
+import matplotlib.pyplot as plt
+import numpy as np
+import math
+from sklearn.metrics import roc_auc_score
+import os.path
+
+
+class NonMyopicEarlyClassification(BaseEstimator, ClassifierMixin):
+    """Early Classification modelling for time series using the model presented in [1]
+
+    Parameters
+    ----------
+    number_cluster : int
+        Number of clusters to form.
+
+    solver :
+        The method of gradient descent used during training
+
+    hidden_layer_sizes : tuple of int
+        Size of the hidden layers
+
+    random_state : int
+        Seed of the MLP
+
+    maximum_iteration : int
+        Maximum number of iterations performed by the MLP
+
+    minimum_time_stamp : int
+        Earliest time at which a classification can be performed on a time series
+
+    lamb : float
+        Value of the hyper parameter lambda
```

Comment:
Maybe your class description should be richer so that it is clear to the end user what `lambda` is.